x-pack/plugins/fleet/server/plugin.ts

```diff
@@ -186,15 +180,15 @@ export class IngestManagerPlugin
     if (deps.features) {
       deps.features.registerKibanaFeature({
         id: PLUGIN_ID,
```

Comment:
__question:__ @legrego IIRC changing feature ID is a breaking change (we changed the `PLUGIN_ID`) and we shouldn't make it in a minor upgrade, but I'm not sure what is our policy regarding features that belong to plugins that are in a beta?

Reply:
Heya Fleet team:

**tl;dr** I think we can rename in this specific case, please take the time to read this entire comment to understand the implications.


> IIRC changing feature ID is a breaking change (we changed the PLUGIN_ID) and we shouldn't make it in a minor upgrade

That's exactly right. Changing a feature id will change the associated privilege ids that are registered with Elasticsearch on startup. For example, prior to this change, a user may have been assigned a custom role which grants access to Ingest Manager. The role would look something like this:

```ts
{
  "name": "some-custom-role",
  "cluster": [],
  "index": [],
  "run_as": [],
  "applications": [
     {
       "application": "kibana-.kibana",
       "privileges": [
         "feature_ingestManager.all"
       ],
       "resources": [
         "space:default"
       ]
     }
   ]
}
```

The privilege that gets assigned to the role (`feature_ingestManager.all`) is derived in part from the feature id. Renaming your feature means that any role which previously granted access to your feature will no longer grant access, because Kibana doesn't have a feature called `ingestManager` anymore. The role should instead look like:

```ts
{
  "name": "some-custom-role",
  "cluster": [],
  "index": [],
  "run_as": [],
  "applications": [
     {
       "application": "kibana-.kibana",
       "privileges": [
         "feature_fleet.all"
       ],
       "resources": [
         "space:default"
       ]
     }
   ]
}
```

The problem is that Kibana can't just go and rewrite all of these role definitions. We don't have any sort of migration system in place yet (https://github.com/elastic/kibana/issues/68814)

> I'm not sure what is our policy regarding features that belong to plugins that are in a beta?

This might be our saving grace. I think we can allow the rename in this specific case, but only because this is a beta feature, and we explicitly tell users not to run this in production yet.

**However, this is something that you need to document in the release notes.** We have to at least give administrators a fighting chance of restoring access to their users, and not surprise them.

Reply:
I will add a release note 👍 , 
I do not think is so problematic here as an user is not able to use the `Fleet` plugin if he do not have the `superuser` role

Reply:
Sounds like y'all got all of the angles covered here, but I just wanted to mention that from various sources it looks like people are very positive about Fleet and Elastic Agent, and that Beta testers go out on a limb to try things out and give us early feedback. So as a general rule, I think it's important to have empathy for them and avoid introducing breaking changes that will essentially surprise and punish people who are trying to help us, if at all possible.

Reply:
I added the release note